apps/dashboard/config/application.rb

```diff
@@ -31,6 +31,7 @@ class Application < Rails::Application
     #
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
+    config.autoload_paths << Rails.root.join("lib/support")
```

Comment:
Can we shift stuff around out of `lib/`.  I think the ActiveModel can move to `models/`, the mailer to `mailers/` and the service logic can likely be encapsulated into one of the two.

Reply:
I added `lib/support` to the `autoload_paths` as a way of allowing new delivery mechanism for the support ticket.
This will allows to drop a new service classes for the RT API into this folder and use it without having to request/submit a new feature to OSC.

The feature has been designed with this idea in mind. The default implementation is email, but it can customized to other delivery mechanisms This is why I have included overrides for the UI template and the service class already in the configuration.

How do you propose that we do the integration with the RT API?

In the meantime, I will move the email implementation classes into the core Rail folder structure as you suggested.

Reply:
> How do you propose that we do the integration with the RT API?

I don't follow why we'd need autoload support.  Most of what you say makes sense I just don't know why any of this functionality needs to be autoloaded.

Or is it just that directory seemed to be the best fit?

Reply:
I just thought that the implementation of a new delivery service for the RT ticket functionality was better suited for the `lib` folder. As it will be a a set of files added to the OOD installation. They could be added into the existing folder structure too, nested under `app`. Under `lib` seemed like a good separation.

I will remove autoload statement.

Reply:
Oh I see, you want to drop in RT support _after the fact_ like a plugin. I guess I assumed you'd push that back to us too. OK, I've never thought of plugin support...

Now I see why you want to configure by the class name directly.